pkg/x509_test.go

```diff
@@ -57,11 +57,15 @@ func (n poolCertVerifier) GetCertificates(i [][]*x509.Certificate, t time.Time,
 }
 
 func (n poolCertVerifier) Verify(cert *x509.Certificate, moment time.Time) error {
+	_, err := n.VerifiedChain(cert, moment)
+	return err
+}
+
+func (n poolCertVerifier) VerifiedChain(cert *x509.Certificate, moment time.Time) ([][]*x509.Certificate, error) {
 	if n.pool == nil {
-		return nil
+		return nil, nil
```

Comment:
returning nil result but also a nil error feels odd (I generally expect an error when the result is nil unless it is a normal function flow). Can pool actually be nil? Maybe returning an error would be better?

Reply:
it's in test. But added an error anyway

Reply:
Oh doh didn't notice 

🤦